py-polars/polars/utils/udfs.py

```diff
@@ -392,19 +395,58 @@ def __iter__(self) -> Iterator[Instruction]:
     def __getitem__(self, item: Any) -> Instruction:
         return self._rewritten_instructions[item]
 
-    def _apply_rules(self, instructions: Iterator[Instruction]) -> list[Instruction]:
+    def _matches(
+        self,
+        idx: int,
+        *,
+        opnames: list[str],
+        argvals: list[set[Any] | dict[Any, Any]] | None,
+    ) -> list[Instruction]:
+        """
+        Check if a sequence of Instructions matches the specified ops/argvals.
+
+        Parameters
+        ----------
+        idx
+            The index of the first instruction to check.
+        opnames
+            The full opname sequence that defines a match.
+        argvals
+            Associated argvals that must also match (in same position as opnames).
+        """
+        n_required_ops, argvals = len(opnames), argvals or []
+        instructions = self._instructions[idx : idx + n_required_ops]
+        if len(instructions) == n_required_ops and all(
+            (
+                inst.opname == match_opname
+                if match_opname[-1] != "*"
+                else inst.opname.startswith(match_opname[:-1])
+            )
+            and (match_argval is None or inst.argval in match_argval)
+            for inst, match_opname, match_argval in zip_longest(
+                instructions, opnames, argvals
+            )
+        ):
+            return instructions
+        return []
+
+    def _rewrite(self, instructions: Iterator[Instruction]) -> list[Instruction]:
         """
         Apply rewrite rules, potentially injecting synthetic operations.
 
         Rules operate on the instruction stream and can examine/modify
         it as needed, pushing updates into "updated_instructions" and
         returning True/False to indicate if any changes were made.
         """
-        self._instructions = instructions
+        self._instructions = list(instructions)
         updated_instructions: list[Instruction] = []
-        for inst in instructions:
-            if not any(
-                apply_rewrite(inst, updated_instructions)
+        idx = 0
+        while idx < len(self._instructions):
+            inst, increment = self._instructions[idx], 1
+            if inst.opname.startswith("LOAD_") and inst.opname not in _LOAD_OPS:
+                x = 1
```

Comment:
what's x?

Reply:
Me forgetting to take out a debugging line - fixed! 🤣